### PR TITLE
Don't suggest turning crate-level attributes into outer style

### LINKED ISSUE #127930: don't suggest turning crate-level attributes into outer style
### Code

```Rust
#![allow(dead_code)]
fn foo() {}
#![feature(iter_array_chunks)]
fn bar() {}
```

### Current output

```Shell
Compiling playground v0.0.1 (/playground)
error: an inner attribute is not permitted in this context
 --> src/lib.rs:3:1
  |
3 | #![feature(iter_array_chunks)]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
4 | fn bar() {}
  | ----------- the inner attribute doesn't annotate this function
  |
  = note: inner attributes, like `#![no_std]`, annotate the item enclosing them, and are usually found at the beginning of source files
help: to annotate the function, change the attribute from inner to outer style
  |
3 - #![feature(iter_array_chunks)]
3 + #[feature(iter_array_chunks)]
  |
```

### Desired output

```Shell
Compiling playground v0.0.1 (/playground)
error: an inner attribute is not permitted in this context
 --> src/lib.rs:3:1
  |
3 | #![feature(iter_array_chunks)]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
4 | fn bar() {}
  | ----------- the inner attribute doesn't annotate this function
  |
  = note: inner attributes, like `#![no_std]`, annotate the item enclosing them, and are usually found at the beginning of source files
```

### Rationale and extra context

reduces distracting noise

### Other cases

_No response_

### Rust Version

```Shell
rustc 1.80.0-nightly (debd22da6 2024-05-29)
binary: rustc
commit-hash: debd22da66cfa97c74040ebf68e420672ac8560e
commit-date: 2024-05-29
host: x86_64-unknown-linux-gnu
release: 1.80.0-nightly
LLVM version: 18.1.6
```

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by chenyukang**
Fixes #127930

### CODE REVIEW

File: compiler/rustc_ast/src/ast.rs [hunk 1]
```diff
@@ -2898,6 +2898,17 @@ pub struct AttrItem {
     pub tokens: Option<LazyAttrTokenStream>,
 }
 
+impl AttrItem {
+    pub fn is_valid_for_inner_outer_sytle(&self) -> bool {
+        self.path == sym::cfg_attr
```
Comment (chenyukang) [inline compiler/rustc_ast/src/ast.rs:2903]: maybe we should try to use `BUILTIN_ATTRIBUTE_MAP` to query.

File: compiler/rustc_parse/src/parser/attr.rs [hunk 2]
```diff
@@ -25,6 +25,7 @@ pub enum InnerAttrForbiddenReason {
     AfterOuterAttribute { prev_outer_attr_sp: Span },
 }
 
+#[derive(PartialEq)]
```
Comment (cjgillot) [inline compiler/rustc_parse/src/parser/attr.rs:28]: Where is this used?

Comment (chenyukang) [reply compiler/rustc_parse/src/parser/attr.rs]: was left unintentionally.
I removed it now.

**Pull request merged**